CRM/Member/Form/Membership.php

```diff
@@ -1488,6 +1441,9 @@ public function submit() {
       );
       $params['source'] = $formValues['source'] ? $formValues['source'] : $params['contribution_source'];
       $params['trxn_id'] = CRM_Utils_Array::value('trxn_id', $result);
+      $params['payment_instrument_id'] = 1;
```

Comment:
This line needs to go - hard coding! The payment processor normally determines the default (it's stored on the processor).